docs/source/model_definition.md

```diff
@@ -67,16 +117,25 @@ model_body = SequentialBlock(
 )
 ```
 
+## Prediction Head (Output Block)
+
+Following the input and transformer blocks, the model outputs its predictions.
+Transformers4Rec supports the following prediction heads, which can have multiple losses and can be combined for multi-task learning and multiple metrics:
+
+- **Next Item Prediction**: Predicts next items for a given sequence of interactions.
+During training, the prediction can be the next item or randomly selected items depending on the masking scheme.
+For inference, the intended purpose is to always predict the next interacted item.
+Cross-entropy and pairwise losses are supported.
 
-## Prediction head (Output Block)
-Following the input and transformer blocks the model outputs its predictions.  The library supports the following prediction heads which can have multiple losses and can be combined for multi-task learning and multiple metrics.
+- **Classification**: Predicts a categorical feature using the whole sequence.
+In the context of recommendation, you can use classification to predict the user's next action such as whether the user will abandon a product in their cart or proceed with the purchase.
 
-- **Next Item Prediction** - Predicts next items for a given sequence of interactions. During training it can be the next item or randomly selected items, depending on the masking scheme. For inference it is meant to always predict the next interacted item. Currently cross-entropy and pairwise losses are supported.
-- **Classification** - Predicts a categorical feature using the whole sequence. In the context of recommendation, which can be used to predict for example if the user is going to abandon a product added to cart or proceed to its purchase.
-- **Regression** - Predicts a continuous feature using the whole sequence, for example the elapsed time until the user returns to a service.
+- **Regression**: Predicts a continuous feature using the whole sequence, such as the elapsed time until the user returns to a service.
 
-In the following example we instantiate a head with the pre-defined `model_body` for the `NextItemPredictionTask`. That head enables the `weight_tying` option, which is described in the next section.
-Decoupling model bodies and heads allow for a flexible model architecture definition, as it allows for multiple towers and/or heads. Finally, the `Model` class combines the heads and wraps the whole model.
+In the following example, a head is instantiated with the predefined `model_body` for the {class}`NextItemPredictionTask <transformers4rec.torch.NextItemPredictionTask>`.
+This head enables the `weight_tying` option.
+Decoupling the model bodies and heads provides a flexible architecture that enables you to define a model with features like multiple towers and multiple heads.
+Lastly, the {class}`Model <transformers4rec.torch.Model>` class combines the heads and wraps the whole model.
 
 ```python
 from transformers4rec.torch import Head, Model
```

Comment:
The new refactor of T4Rec API removed the flag `hf_format` from the `NextItemPredictionTask` class. So we should update the code snippet to use ```NextItemPredictionTask(weight_tying=True)```